### PR TITLE
add toggle button as new v3 web component

### PR DESCRIPTION
**Pull request by chrisdholt**
## Related Issue(s)

- Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/14df67d349d979290b8a802b12b14d0bb66d9441'>14df67d349d979290b8a802b12b14d0bb66d9441</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/web-components/src/toggle-button/toggle-button.styles.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    border-color: ${colorNeutralStroke1};
+    background-color: ${colorNeutralBackground1Selected};
+    color: ${colorNeutralForeground1};
+    border-width: ${strokeWidthThin};
+  }
+
+  :host([aria-pressed='true']:hover) .control {
+    border-color: ${colorNeutralStroke1Hover};
+    background-color: ${colorNeutralBackground1Hover};
+    color: ${colorNeutralForeground1};
```
Comment (eljefe223) [inline packages/web-components/src/toggle-button/toggle-button.styles.ts:41]: Do we need to set color here? Specificity issue?

File: packages/web-components/src/toggle-button/toggle-button.styles.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
+import { css } from '@microsoft/fast-element';
+import { styles as ButtonStyles } from '../button/button.styles.js';
```
Comment (eljefe223) [inline packages/web-components/src/toggle-button/toggle-button.styles.ts:2]: Does anchor use these styles also? Should we have a generic location and a more generic name?

Comment (chrisdholt) [reply packages/web-components/src/toggle-button/toggle-button.styles.ts:2]: It's technically the button styles and these are all subtle variations of button. I'm not sure a deeper extraction is needed here yet. I'd much prefer we start here and further abstract if we need to.

File: packages/web-components/src/toggle-button/toggle-button.styles.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  :host([aria-pressed='true'][appearance='primary']:hover) .control {
+    background-color: ${colorBrandBackgroundHover};
+  }
+
+  :host([aria-pressed='true'][appearance='primary']:active) .control {
+    background-color: ${colorBrandBackgroundPressed};
+  }
+
+  :host([aria-pressed='true'][appearance='subtle']) .control {
+    --button-border-color: transparent;
```
Comment (eljefe223) [inline packages/web-components/src/toggle-button/toggle-button.styles.ts:65]: Seems like some places we are just setting the border color and here we are reassigning a variable. Probably should be consistent

Comment (chrisdholt) [reply packages/web-components/src/toggle-button/toggle-button.styles.ts:65]: This is wrong - should just be assigning it. Great catch here.

File: packages/web-components/src/toggle-button/toggle-button.styles.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  }
+
+  :host([aria-pressed='true'][appearance='primary']:active) .control {
+    background-color: ${colorBrandBackgroundPressed};
+  }
+
+  :host([aria-pressed='true'][appearance='subtle']) .control {
+    border-color: transparent;
+    background-color: ${colorSubtleBackgroundSelected};
+    color: ${colorNeutralForeground2BrandSelected};
```
Comment (miroslavstastny) [inline packages/web-components/src/toggle-button/toggle-button.styles.ts:65]: According to design spec, content color in `subtle pressed` should use colorNeutralForeground2Selected (no brand).
Only the icon should have the brand color.

File: packages/web-components/src/toggle-button/toggle-button.stories.ts [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+export const Shape = renderComponent(html<ToggleButtonStoryArgs>`
+  <fluent-toggle-button shape="rounded">Rounded</fluent-toggle-button>
+  <fluent-toggle-button shape="circular">Circular</fluent-toggle-button>
+  <fluent-toggle-button shape="square">Square</fluent-toggle-button>
+`);
+
+export const Size = renderComponent(html<ToggleButtonStoryArgs>`
+  <fluent-toggle-button size="small">Small</fluent-toggle-button>
+  <fluent-toggle-button size="small" icon
```
Comment (miroslavstastny) [inline packages/web-components/src/toggle-button/toggle-button.stories.ts:108]: There is no `icon` attribute (multiple places)

**Comment by chrisdholt** [comment]
> Toggle button design spec prescribes how an icon in the toggle button should look like in the checked state. The icon should be filled and in brand color for some appearances. The toggle button implementation does not address that so we should have a clear plan on how we want to support these use cases.

I think this is noted as a TODO across all buttons as some of them do swap icon color on hover as well. I can update the checked state but was unsure if this was a hover ergonomic or a delta where it's "checked".

**Comment by chrisdholt** [comment]
@miroslavstastny all issues addressed - I'm going to go ahead and merge this and if need be I'll jump on any fast follow issues.

**Pull request merged**